include/wx/stc/completionitem.h

```diff
@@ -0,0 +1,25 @@
+#ifndef _WX_STC_COMPLETIONITEM_H_
+#define _WX_STC_COMPLETIONITEM_H_
+
+#include "wx/string.h"
+#include "wx/clntdata.h"
+
+class WXDLLIMPEXP_STC wxCompletionItem
+{
+public:
+    wxCompletionItem(const wxString &name, int icon, const wxString &doc = wxString(), wxClientData *data = NULL);
+    virtual ~wxCompletionItem();
```

Comment:
Is this class supposed to be polymorphic or not? If yes, it should have some virtual functions and it should _not_ use `wxClientData`, being a kind of client data on its own. If not, it shouldn't have a virtual dtor.

I'd prefer to make it polymorphic rather than using client data, which is a kludge used when there is no other possibility.

Reply:
It was not intended to be polymorphic for one reason: the only way I knew to alert the user when item is selected is attaching a wxClient object. 

wxSTC events don't have an option to set payload. How will I post it back selected object on event? Do you have an idea? If I can solve this then it will be polymorphic and client object stuff will be removed.